src/users/user-manager.js

```diff
@@ -1,51 +1,70 @@
-var response = require('../Objects/response.js');
-var userModels = require('./user-models.js');
-var Student = userModels.Student;
-var Admin = userModels.Admin;
-
-var createUser = (reqData, createCallback) => {
-  var user;
-  if (reqData.role == 'Student') {
-    // Create new student.
-    user = new Student({
-      approvalStatus: false,
-      email: reqData.email,
-      firstName: reqData.firstName,
-      lastName: reqData.lastName,
-      points: {
-        career: 0,
-        community: 0,
-        firstother: 0,
-        firstworkshops: 0,
-        mentor: 0,
-        other: 0,
-        outreach: 0,
-        professor: 0,
-        staff: 0,
-        misc: 0
-      },
-      role: 'Student',
-      studentId: reqData.studentId
-    });
-
-  } else if (reqData.role == 'Admin') {
-    // Create new admin.
-    user = new Admin({
-      email: reqData.email,
-      firstName: reqData.firstName,
-      lastName: reqData.lastName,
-      role: 'Admin'
-    });
+const response = require('../Objects/response.js');
+const userModels = require('./user-models.js');
+const User = userModels.User;
+const Student = userModels.Student;
+const Admin = userModels.Admin;
 
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function createResponse
+ * @param {Object} err - The error.
+ * @param {Number} id - The user UID.
+ */
+
+/**
+ * Given User information, create a User.
+ * @param {Object} data - The request body data.
+ * @param {String} data.role - The user role.
+ * @param {String} data.email - The user email address.
+ * @param {String} data.password - The user password.
+ * @param {createResponse} next - The callback function to run after this function
+  *     finishes.
+ */
+const createUser = (data, next) => {
+  // Ensure the required data is available.
+  if (!data || !data.role || !data.email || !data.password ||
```

Comment:
Since these are required fields in the schema, I think mongoose will pass an error if any of these are undefined in a document being saved.  I.e., we probably just need to check ```!data```, and the others should be checked by mongoose during document validation during save.